### DISCUSSION TITLE
Can I use renderReadableStream?

### DISCUSSION THREAD

**Discussion by nickytonline** [netlify/remix-template#100] [Q&A]
Yes, you can use [renderReadableStream](https://beta.reactjs.org/reference/react-dom/server/renderToReadableStream). It was in the Netlify Edge Functions template initially before #101. Still, it made using the template difficult from a DX standpoint.

You had to first run `netlify build,` and then you could run `netlify dev` to develop your site. See #92. The developer experience is much better now, where you can run `netlify dev` and make changes as you would expect.

However, this means we had to revert to [renderToString](https://beta.reactjs.org/reference/react-dom/server/renderToString) because the Remix development server is running Node.js which does not support [Web Streams](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API), what `renderToReadableStream` uses.

You can replace your `entry.server.server.tsx` file with the one in https://github.com/netlify/remix-template/commit/fd3139b9458589df7e31ead8cd353d219452400b. 

The development flow would be to run:
1. `npx remix watch`
2. `netlify build`
3. `netlify dev`
4. Make changes and hard refresh the browser.

To be clear, this is not a limitation of the Netlify CLI. It's due to a lack of Web Streams support in Node.js, as the `renderToReadableStream` docs mention, which the Remix dev server runs on.

**Comment by jahvi** [comment]
Doesn't Node 18 support web streams now? If so could the flow be improved in that version?
https://nodejs.dev/en/api/v18/webstreams/

**Comment by nickytonline** [comment]
@jahvi, I saw that and upgraded ran the project to use Node 18 but still ran into errors about WebStreams not being supported. I'm going to take peek at this again as I wonder if maybe the Remix dev server, what we're running when `netlify dev` is running, is not supporting it. 🤔

**Comment by nickytonline** [comment]
See https://github.com/facebook/react/issues/24226 @jahvi. I may move the template back to use `renderReadableStream`. Just looking how to provide good DX for it is all.
